# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/loader/WorkerLoadContext.h
# Commit: 1034aeb0f8e2
# Full Hash: 1034aeb0f8e2d86ea04d8e6a8c717f13c9d772eb
# Author: Yulia Startsev <ystartsev@mozilla.com>
# Date: 2022-11-18 15:46:32
# Regressor Bug: 1800496
# File Overlap Count: 2
# Description:
#   Bug 1800496 - Revert mCacheCreator to being defined as part of ScriptLoaderRunnable; r=asuth
#   
#   Previously, in the cancellation refactor, we removed knowledge of the list of scripts being loaded on the main thread. As a result of this, we could no longer clear the cache creator on the main thread after iterating over all requests. We can now do that once again, and it simplifies the code.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D162220
# ==============================================================================

diff -r dfe78c41e4eb -r 1034aeb0f8e2 dom/workers/loader/WorkerLoadContext.h
--- a/dom/workers/loader/WorkerLoadContext.h	Fri Nov 18 10:02:28 2022 +0000
+++ b/dom/workers/loader/WorkerLoadContext.h	Fri Nov 18 10:02:28 2022 +0000
@@ -133,15 +133,6 @@
   // when we're going to have an mCachePromise.
   nsCOMPtr<nsIInputStream> mCacheReadStream;
 
-  nsMainThreadPtrHandle<workerinternals::loader::CacheCreator> mCacheCreator;
-
-  void ClearCacheCreator();
-
-  void SetCacheCreator(
-      RefPtr<workerinternals::loader::CacheCreator> aCacheCreator);
-
-  RefPtr<workerinternals::loader::CacheCreator> GetCacheCreator();
-
   enum CacheStatus {
     // By default a normal script is just loaded from the network. But for
     // ServiceWorkers, we have to check if the cache contains the script and
@@ -197,6 +188,8 @@
 
   already_AddRefed<JS::loader::ScriptLoadRequest> ReleaseRequest();
 
+  workerinternals::loader::CacheCreator* GetCacheCreator();
+
   RefPtr<workerinternals::loader::ScriptLoaderRunnable> mRunnable;
 
   bool mExecutionScheduled = false;
